### PR TITLE
Bug 1916373: upi: allow IPsec ports

### PR DESCRIPTION
**Pull request by markdgray**

### CODE REVIEW

File: upi/aws/cloudformation/03_cluster_security.yaml [hunk 1]
```diff
@@ -274,6 +330,62 @@ Resources:
...
+
+  WorkerIngressIpsecEsp:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !GetAtt WorkerSecurityGroup.GroupId
+      SourceSecurityGroupId: !GetAtt WorkerSecurityGroup.GroupId
+      Description: IPsec ESP packets
+      IpProtocol: esp
+
+  WorkerIngressMasterIpsecIke:
```
Comment (staebler) [inline upi/aws/cloudformation/03_cluster_security.yaml:361]: I don't see master-to-worker security group rules in the IPI flow.

**Review by staebler — COMMENTED** [review]
There is one discrepancy between the IPI and UPI. The rest looks good.

File: upi/aws/cloudformation/03_cluster_security.yaml (hunk 1 above)
Comment (markdgray) [reply upi/aws/cloudformation/03_cluster_security.yaml:361]: uggh .. that looks like a bug in the IPI flows. I need to update them and raise another Bz. Good catch.

File: upi/openstack/security-groups.yaml [hunk 2]
```diff
@@ -93,6 +93,22 @@
...
+      security_group: "{{ os_sg_master }}"
+      protocol: udp
+      remote_ip_prefix: "{{ os_subnet_range }}"
+      port_range_min: 500
+      port_range_max: 500
+
+  - name: 'Create master-sg rule "IPsec NAT-T"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_master }}"
+      protocol: udp
```
Comment (markdgray) [inline upi/openstack/security-groups.yaml:107]: Just to note that, here, I was unable to add "esp" as a protocol. That looks like an issue with https://docs.ansible.com/ansible/2.10/collections/openstack/cloud/security_group_rule_module.html. I raised a Bugzilla against this: https://bugzilla.redhat.com/show_bug.cgi?id=1916474

File: upi/aws/cloudformation/03_cluster_security.yaml (hunk 1 above)
Comment (markdgray) [reply upi/aws/cloudformation/03_cluster_security.yaml:361]: https://github.com/openshift/installer/pull/4554

File: upi/openstack/security-groups.yaml (hunk 2 above)
Comment (markdgray) [reply upi/openstack/security-groups.yaml:107]: However, if you know a way to do this, please let me know.

Comment (staebler) [reply upi/openstack/security-groups.yaml:107]: I'm afraid that I do not. I don't know too much about OpenStack.

**Review by staebler — COMMENTED** [review]
/approve
/lgtm

Hold for passing UPI e2e tests.
/hold

**Comment by markdgray** [comment]
The failures look to be unrelated flakes. 

/test e2e-aws-upi
/test e2e-gcp-upi
/test e2e-openstack-upi

**Comment by markdgray** [comment]
timeouts during install "Get "https://registry.build02.ci.openshift.org/openshift/token?account=user&scope=repository%3Aci-op-r8d57l95%2Fstable%3Apull": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"

/retest

**Comment by markdgray** [comment]
@staebler Do these look like known issues or could it be a bug in the rules? I am not familiar with these build jobs?

**Comment by staebler** [comment]
> @staebler Do these look like known issues or could it be a bug in the rules? I am not familiar with these build jobs?

The e2e-aws-upi test is failing on creating the resources, but from the history of that test it is not evident that it is related to the changes in this PR. I would still like to get at least a successful install, though.
/test e2e-aws-upi

The e2e-gcp-upi test looks like it is broken missing a dependency. This is not related to the changes in this PR. It has been broken since at least July.
```
CommandException: The signurl command requires the pyopenssl library (try pip install pyopenssl or easy_install pyopenssl)
```

The e2e-openshift-upi test did have one green pass and is at least installing on other runs. I am satisfied with that.

**Pull request merged**